plugins/inputs/dnac/dnac.go

```diff
@@ -0,0 +1,176 @@
+	return "a Cisco DNAC plugin"
+}
+
+func (d *Dnac) SampleConfig() string {
+	return `
+  ## Specify DNAC Base URL
+  dnacbaseurl = "https://sandboxdnac.cisco.com"
+  ## Specify Credentials
+  username = "devnetuser"
+  password = "Cisco123!"
+  ## SSL Verify true/false
+  sslverify = "false"
+  ## Health types to report
+  report = ["client","network"]
+`
+}
+
+// Init is for setup, and validating config.
+func (d *Dnac) Init() error {
+	err := d.InitClient()
+	if err != nil {
+		return err
+	}
+	err = d.Client.AuthClient()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (d *Dnac) InitClient() error {
+	var debug string
+	if telegraf.Debug {
+		debug = "true"
+	} else {
+		debug = "false"
+	}
+
+	var client, err = dnac_sdk.NewClientWithOptionsNoAuth(d.DnacBaseURL, d.Username, d.Password, debug, d.SSLVerify)
+	if err != nil {
+		d.Log.Errorf("Connection or login to DNAC failed")
+		return err
+	}
+	d.Client = client
+	return nil
+}
+
+func NewDnac() *Dnac {
+	return &Dnac{}
+}
+
+func (d *Dnac) Gather(acc telegraf.Accumulator) error {
+	var err error
+
+	var dnacURL, urlErr = url.Parse(d.DnacBaseURL)
+	if urlErr != nil {
+		d.Log.Errorf("Invalid DNAC Base URL Provided")
+		return urlErr
+	}
+
+	if d.Client == nil {
+		err = d.InitClient()
+		if err != nil {
+			return err
+		}
+	}
+
+	err = d.Client.AuthClient()
+	if err != nil {
+		return err
+	}
+
+	for healthType := range d.Report {
+		if d.Report[healthType] == "client" {
+			getOverallClientHealthQueryParams := &dnac_sdk.GetOverallClientHealthQueryParams{}
+			clientHealth, _, err := d.Client.Clients.GetOverallClientHealth(getOverallClientHealthQueryParams)
+			if err != nil {
+				d.Log.Errorf("Client health request failed")
+				return err
+			}
+			for _, response := range *clientHealth.Response {
+				tags := map[string]string{
+					"host":    dnacURL.Host,
+					"site_id": response.SiteID,
+				}
+				fields := make(map[string]interface{})
+				for _, clientType := range *response.ScoreDetail {
+					l1Prefix := clientType.ScoreCategory.ScoreCategory + "_" + clientType.ScoreCategory.Value
+					fields[internal.SnakeCase(l1Prefix+"_client_count")] = clientType.ClientCount
+					fields[internal.SnakeCase(l1Prefix+"_score_value")] = clientType.ScoreValue
+					if clientType.ScoreList != nil {
+						for _, scoreType := range *clientType.ScoreList {
+							l2Prefix := scoreType.ScoreCategory.ScoreCategory + "_" + scoreType.ScoreCategory.Value
+							fields[internal.SnakeCase(l1Prefix+"_"+l2Prefix+"_client_count")] = scoreType.ClientCount
+							if scoreType.ScoreList != nil {
+								for _, rootCause := range *scoreType.ScoreList {
+									if rootCause.ScoreCategory.ScoreCategory == "rootCause" {
+										l3Prefix := rootCause.ScoreCategory.ScoreCategory + "_" + rootCause.ScoreCategory.Value
+										fields[internal.SnakeCase(l1Prefix+"_"+l2Prefix+"_"+l3Prefix+"_client_count")] = rootCause.ClientCount
+									}
+								}
+							}
+						}
+					}
```

Comment:
Can you please rewrite this so it isn't so nested? If you checked for `clientType.ScoreList == nil` instead then continue on that case it will remove some layers of indentation.

Reply:
I denested both checks for nil by using continue.